internal/framework/service/leaked_credential_check_rules/resource.go

```diff
@@ -0,0 +1,230 @@
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*muxclient.Client)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"unexpected resource configure type",
+			fmt.Sprintf("Expected *muxclient.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *LeakedCredentialCheckRulesResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan LeakedCredentialCheckRulesModel
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	zoneID := plan.ZoneID.ValueString()
+	var createdRules []LCCRuleValueModel
+	if len(plan.Rules) > 0 {
+		for _, rule := range plan.Rules {
+			createParam := cloudflare.LeakedCredentialCheckCreateDetectionParams{
+				Username: rule.Username.ValueString(),
+				Password: rule.Password.ValueString(),
+			}
+			detection, err := r.client.V1.LeakedCredentialCheckCreateDetection(ctx, cloudflare.ZoneIdentifier(zoneID), createParam)
+			if err != nil {
+				resp.Diagnostics.AddError("Error creating a user-defined detection patter for Leaked Credential Check", err.Error())
+				return
+			}
+			createdRules = append(createdRules, LCCRuleValueModel{ID: types.StringValue(detection.ID), Username: rule.Username, Password: rule.Password})
+		}
+
+	}
+	plan.Rules = createdRules
+
+	diags = resp.State.Set(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (r *LeakedCredentialCheckRulesResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state LeakedCredentialCheckRulesModel
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	zoneID := state.ZoneID.ValueString()
+	rules, err := r.client.V1.LeakedCredentialCheckListDetections(ctx, cloudflare.ZoneIdentifier(zoneID), cloudflare.LeakedCredentialCheckListDetectionsParams{})
+	if err != nil {
+		resp.Diagnostics.AddError("Error listing Leaked Credential Check user-defined detection patterns", err.Error())
+		return
+	}
+	var readRules []LCCRuleValueModel
+	for _, rule := range rules {
+		readRules = append(readRules, buildLCCRuleValueModel(rule))
+	}
+	state.Rules = readRules
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (r *LeakedCredentialCheckRulesResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var plan LeakedCredentialCheckRulesModel
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	zoneID := cloudflare.ZoneIdentifier(plan.ZoneID.ValueString())
+	// fetch existing rules from API
+	existing_rules, err := r.client.V1.LeakedCredentialCheckListDetections(ctx, zoneID, cloudflare.LeakedCredentialCheckListDetectionsParams{})
```

Comment:
this functionality isn't going to be ported to v5 as it is custom handling of the payloads. is there a reason this cannot be achieved using the schema and remote service?